src/flick/lst/views.py

```diff
@@ -1,3 +1,184 @@
+        lst = Lst.objects.get(pk=pk)
+        profile = Profile.objects.get(user=request.user)
+        if not profile:
+            return failure_response(f"No user to be found with id of {request.user.id}.")
+        user_is_collaborator = profile in lst.collaborators.all()
+        user_is_owner = profile == lst.owner
+        if not lst.is_private or user_is_collaborator or user_is_owner:
+            serializer = LstSerializer(lst, many=False)
+            return success_response(serializer.data)
+        else:
+            return failure_response(
+                f"User of id {request.user.id} does not have the right permissions to view list of id {pk}."
+            )
+
+    def delete(self, request, pk):
+        """
+        Delete a list by id.
+        Only owners of lists can delete their lists.
+        """
+        if not Lst.objects.filter(pk=pk):
+            return failure_response(f"List of id {pk} does not exist.")
+        lst = Lst.objects.get(pk=pk)
+        profile = Profile.objects.get(user=request.user)
+        if not profile == lst.owner:
+            return failure_response(f"User of id {request.user.id} is not the owner of list of id {pk}.")
+        lst.delete()
+        return success_response(f"List of id {pk} has been deleted.")
+
+    def post(self, request, pk):
+        """
+        Update a list by id.
+        Collaborators can update lst_pic, is_favorite, is_watched, collaborators, and shows.
+        An owner can update lst_name, lst_pic, is_favorite, is_private, is_watched, collaborators, the owner (can cede ownership completely to another user), and shows.
+        """
+        data = json.loads(request.body)
+        lst_name = data.get("lst_name")
+        lst_pic = data.get("lst_pic")
+        is_favorite = data.get("is_favorite", False)
+        is_private = data.get("is_private", False)
+        is_watched = data.get("is_watched", False)
+        collaborator_ids = data.get("collaborators", [])
+        owner_id = data.get("owner", request.user.id)
+        show_ids = data.get("shows", [])
+
+        if not Lst.objects.filter(pk=pk):
+            return failure_response(f"No list to be found with id of {pk}.")
+        lst = Lst.objects.get(pk=pk)
+
+        profile = Profile.objects.get(user=request.user)
+        if not profile:
+            return failure_response(f"No user to be found with id of {request.user.id}.")
+
+        user_is_collaborator = profile in lst.collaborators.all()
+        user_is_owner = profile == lst.owner
+
+        if not (user_is_collaborator or user_is_owner):
+            return failure_response(f"User {request.user} does not have the credentials to list of id {pk}.")
+        if user_is_collaborator:
+            lst.lst_pic = lst_pic
+            lst.is_favorite = is_favorite
+            lst.is_watched = is_watched
+            lst.collaborators.clear()
+            for c_id in collaborator_ids:
```

Comment:
should collaborators be able to delete another collaborator?

Reply:
oh that's a good question... @cindyh25